packages/charts/chart-utilities/src/PlotlySchemaConverter.ts

```diff
@@ -538,8 +538,13 @@ export const mapFluentChart = (input: any): OutputChartType => {
           const scatterData = traceData as Partial<PlotData>;
           const isAreaChart = isScatterAreaChart(scatterData);
           const isScatterChart = isScatterMarkers(scatterData.mode ?? '');
+          const hasLineShape =
+            Array.isArray(validSchema?.layout?.shapes) &&
+            validSchema.layout.shapes.some((shape: any) => shape.type === 'line');
+          const isStringY = isStringArray(scatterData.y);
+
           if (isScatterChart) {
-            return { isValid: true, traceIndex, type: 'scatter' };
+            return { isValid: true, traceIndex, type: hasLineShape && !isStringY ? 'line' : 'scatter' };
```

Comment:
line does not support string x. Create a function for conditions in line 672 and use that here.

Reply:
The whole doesScatterNeedFallback is not needed. Only condition in line 672. Dont create duplicate code for same condition

Reply:
Pass hasLineShape as parameter to doesScatterNeedFallback and add it to the first check